### ISSUE TITLE
Actions: Variable substitution in commented lines

### ISSUE THREAD

**Issue by girishenoy2003** [actions/runner#1486]
If there are any variables in the commented lines get substituted in the logs of action workflow. 

**To Reproduce**

1.  Write Any simple action work flow
2. In a job write following step
```
 - name: Build App
    run: |
          #cd ${{ github.workspace }}
```
3. Navigate to the logs of the step and expand the command. You can see the following
```
#cd /home/user/actions-runner/_work/cd-demo-app/cd-demo-app
```

Screenshot:
![image](https://user-images.githubusercontent.com/21303133/141755282-31108009-6495-4ccb-b40c-18f94cf41bc9.png)

**Expected behavior**
As the line was commented, it is expected not to substitute the variable. But the variable is getting substituted in the logs as shown in the snapshot

## Runner Version and Platform
Runner version - 2.279.0
Running on Linux - x64

**Comment by TingluoHuang** [comment]
@girishenoy2003 
` ${{ github.workspace }}` is not a variable, it's an expression syntax for GitHub Actions, it will get evaluated and expanded before executing any script.
https://docs.github.com/en/actions/learn-github-actions/expressions#about-expressions

**Comment by girishenoy2003** [comment]
@TingluoHuang 
Agreed that it is an expression syntax but evaluating such expressions on the commented lines is expected?

**Comment by TingluoHuang** [comment]
The runner doesn't know it's a comment for a given shell. The runner treat all inputs as string.

**Comment by girishenoy2003** [comment]
That is exactly my initial query was... Isn't that runner should differentiate the commented lines?
Or
There is no concept of comments in action?

**Comment by TingluoHuang** [comment]
There is no concept of comments since we don't know what is the syntax for comment in all these shells
ex:
- `#` in bash
- `<#` in powershell
-  `rem` in batch
